### PR TITLE
Update DelayedAutoCancellable

### PR DESCRIPTION
**Pull request by ddddxxx**

### CODE REVIEW

File: Sources/CXExtensions/DelayedAutoCancellable.swift [hunk 1]
```diff
@@ -7,23 +7,26 @@ import CXFoundation
...
+        // FIXME: we should schedule non-repeatedly, but it's not cancellable.
+        self.scheduleCanceller = scheduler.schedule(after: scheduler.now.advanced(by: interval), interval: .seconds(.max), tolerance: tolerance, options: options) { [unowned self] in
+            self.cancel()
+        }
     }
     
     public func cancel() {
-        self.c.cancel()
+        scheduleCanceller.cancel()
+        canceller.cancel()
```
Comment (luoxiu) [inline Sources/CXExtensions/DelayedAutoCancellable.swift]: I'm thinking if DelayedAutoCancellable should release these cancellers after cancel, just like [AnyCancellable](https://github.com/cx-org/CombineX/blob/95056aaa02a13caa55263ce44368633936af9a1a/Tests/CombineXTests/AnyCancellableSpec.swift#L67). 🤔

Comment (ddddxxx) [reply Sources/CXExtensions/DelayedAutoCancellable.swift]: Done. I'm wondering the type of `scheduleCanceller`. Should it be `Cancellable?` or `(() -> Void)?`

Comment (luoxiu) [reply Sources/CXExtensions/DelayedAutoCancellable.swift]: `Cancellable` would make more sense, and why is `canceller` a closure in the new commit? Two `Cancellable` make it easy to know what we are doing I think.

File: Sources/CXExtensions/DelayedAutoCancellable.swift [hunk 2]
```diff
@@ -7,26 +7,28 @@ import CXFoundation
...
     public func cancel<S: Scheduler>(after interval: S.SchedulerTimeType.Stride, tolerance: S.SchedulerTimeType.Stride = .zero, scheduler: S, options: S.SchedulerOptions? = nil) -> DelayedAutoCancellable {
-        return DelayedAutoCancellable(cancel: self, after: interval, tolerance: tolerance, scheduler: scheduler, options: options)
+        return DelayedAutoCancellable(cancel: self.cancel, after: interval, tolerance: tolerance, scheduler: scheduler, options: options)
     }
 }
 
 public final class DelayedAutoCancellable: Cancellable {
     
-    private let canceller: Cancellable
+    private var cancelBody: (() -> Void)?
```
Comment (luoxiu) [inline Sources/CXExtensions/DelayedAutoCancellable.swift:16]: Why is canceller a closure? Two `Cancellable` make it easy to know what we are doing here.

Comment (ddddxxx) [reply Sources/CXExtensions/DelayedAutoCancellable.swift:16]: Store a protocol type could result heap allocation due to existential container.

Comment (ddddxxx) [reply Sources/CXExtensions/DelayedAutoCancellable.swift:16]: As for scheduleCanceller, the scheduler gives us an existential container anyway, so I just take it.

Comment (luoxiu) [reply Sources/CXExtensions/DelayedAutoCancellable.swift:16]: 🤣, ok, ok.

**Pull request merged**